compiler/src/dotty/tools/dotc/transform/init/Objects.scala

```diff
@@ -470,7 +470,7 @@ object Objects:
 
     /** Store the heap as a mutable field to avoid threading it through the program. */
     class MutableData(private[Heap] var heap: Data):
-      private[Heap] def update(addr: Addr, value: Value): Unit =
+      private[Heap] def writeUnion(addr: Addr, value: Value): Unit =
```

Comment:
I wonder about the name change. Is the `Union` intended to convey that we update the heap with the join? In that case, I suggest `writeJoin` rather than `writeUnion` for consistency.